drivers/interrupt_controller/Kconfig.multilevel

```diff
@@ -0,0 +1,119 @@
+# Kconfig - Multilevel interrupt configuration
+#
+# Copyright (c) 2017 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config MULTI_LEVEL_INTERRUPTS
+	bool "Multi-level Interrupts"
+	default n
+	depends on GEN_SW_ISR_TABLE
+	help
+	  Multiple levels of interrupts are normally used to increase the
+	  number of addressable interrupts in a system. For e.g., if 2 levels
+	  of interrupts are used, the 2nd level controller would combine all
+	  interrupts routed to it into one line which then gets routed to one
+	  of the lines in 1st level interrupt controller.
+
+config MAX_IRQ_PER_AGGREGATOR
+	int "Max IRQs per interrupt aggregator"
+	default 0
+	depends on MULTI_LEVEL_INTERRUPTS
+	help
+	  This represents the max number of interrupts that an aggregator
+	  can map.
+
+config 2ND_LEVEL_INTERRUPTS
+	bool "Second-level Interrupts"
+	default n
+	depends on MULTI_LEVEL_INTERRUPTS
+	help
+	  Second level interrupts are used to increase the number of
+	  addressable interrupts in a system.
+
+config 2ND_LVL_ISR_TBL_OFFSET
+	int "Offset in the SW ISR Table for 2nd level interrupt controller"
+	default 0
+	depends on MULTI_LEVEL_INTERRUPTS
+	help
+	  This is the offset in the SW ISR table beginning where the ISRs for
+	  2nd level interrupts are located. This typically is allocated after
+	  allocating first level interrupts.
+
+config NUM_2ND_LEVEL_AGGREGATORS
```

Comment:
How is this used?

Reply:
This is used to calculate the ISR offsets (implementation will be specific to a particular board). I will now make use of this value for some other purpose also (see my next update for this patchset).